src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -625,13 +635,25 @@ public TResult TryGetLast(out bool found)
                 return sourceFound ? _selector(input) : default(TResult);
             }
 
+            private TResult[] LazyToArray()
```

Comment:
Just checked. `LazyToArray` comes in at an extra 306 bytes of code: https://gist.github.com/jamesqo/6b5f2cc27ca36fd3f488c7be57a87cb4 It would probably be wise to avoid generating all that unnecessary code if it's never called.